src/mqueryfront/src/App.js

```diff
@@ -21,20 +22,32 @@ function getCurrentTokenOrNull() {
 
 function App() {
     const [config, setConfig] = useState(null);
+    const tokenIntervalRef = useRef(null);
 
     useEffect(() => {
         api.get("/server").then((response) => {
             setConfig(response.data);
         });
+        tokenIntervalRef.current = setInterval(() => {
+            refreshAccesToken();
+        }, 900000); // refresh token every 15 minutes just in case user was idle.
+        return () => clearInterval(tokenIntervalRef.current);
     }, []);
-
-    const login = (rawToken) => {
-        localStorage.setItem("rawToken", rawToken);
-        window.location.href = "/";
+    const login = async (token_data) => {
+        token_data.not_before_policy = token_data["not-before-policy"];
+        delete token_data["not-before-policy"];
+        const response = await api.post("/login", token_data);
```

Comment:
```suggestion
        await api.post("/login", token_data);
```
since response is not used anywhere